client-metrics/src/main/scala/org/http4s/client/metrics/Metrics.scala

```diff
@@ -0,0 +1,71 @@
+package org.http4s.client.metrics
+
+import cats.data.Kleisli
+import cats.effect._
+import cats.implicits._
+import com.codahale.metrics.{Counter, MetricRegistry, Timer}
+import java.util.concurrent.TimeUnit
+import org.http4s.{Request, Status}
+import org.http4s.client.{Client, DisposableResponse}
+
+object Metrics {
+  def apply[F[_]: Sync](registry: MetricRegistry, prefix: String = "org.http4s.client")(
+      client: Client[F]): Client[F] = {
+
+    def withMetrics(metrics: MetricsCollection)(req: Request[F]): F[DisposableResponse[F]] =
+      for {
+        start <- Sync[F].delay(System.nanoTime())
+        _ <- Sync[F].delay(metrics.activeRequests.inc())
+        resp <- client.open(req)
+        _ <- Sync[F].delay(metrics.activeRequests.dec())
+        iResp <- Sync[F].delay(instrumentResponse(start, metrics, resp))
+      } yield iResp
+
+    def instrumentResponse(
+        start: Long,
+        metrics: MetricsCollection,
+        disposableResponse: DisposableResponse[F]): DisposableResponse[F] = {
+      val newDisposable = for {
+        elapsed <- Sync[F].delay(System.nanoTime() - start)
```

Comment:
This is measuring time to first byte.  Should we be measuring that _and_ total time?  See #1961 for related discussion.

Reply:
I thought this will be measuring total time, as it's being recorded when disposing the response...

Reply:
I thought this will be measuring total time, as it's being recorded when disposing the response...

Reply:
You're right.  I was looking for a finalizer on the response body, like we do on the server side, but this will work and is the right way to do it.